compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp

```diff
@@ -0,0 +1,208 @@
+  assert(node->dtype() == loco::DataType::S64 || node->dtype() == loco::DataType::S32);
+
+  if (node->dtype() == loco::DataType::S64)
+    return node->at<loco::DataType::S64>(idx);
+  return static_cast<int64_t>(node->at<loco::DataType::S32>(idx));
+}
+
+bool substitute_strided_slice_to_reshape(luci::CircleStridedSlice *ss_node)
+{
+  if (ss_node->shrink_axis_mask() == 0)
+    return false;
+
+  // TODO Consider cases with ellipsis_mask and new_axis_mask
+  // NOT YET SUPPORTED
+  if (ss_node->ellipsis_mask() != 0 or ss_node->new_axis_mask() != 0)
+    return false;
+
+  auto begin_const = dynamic_cast<luci::CircleConst *>(ss_node->begin());
+  auto strides_const = dynamic_cast<luci::CircleConst *>(ss_node->strides());
+  auto end_const = dynamic_cast<luci::CircleConst *>(ss_node->end());
+  auto input_node = loco::must_cast<luci::CircleNode *>(ss_node->input());
+
+  if (not(begin_const && strides_const && end_const && input_node))
+    return false;
+
+  // condition check
+  std::bitset<32> begin_mask(ss_node->begin_mask());
+  std::bitset<32> end_mask(ss_node->end_mask());
+  std::bitset<32> shrink_axis_mask(ss_node->shrink_axis_mask());
+
+  uint input_rank = input_node->rank();
+  for (uint32_t i = 0; i < input_rank; i++)
+  {
+    if (!input_node->dim(i).known())
+      return false;
+
+    auto begin_dim = value_from_circle_const(begin_const, i);
+    if (not(begin_dim == 0 or begin_mask.test(i) == true))
+      return false;
+
+    // NOTE:
+    //    In Tensorflow and TFLite, e.g., if input_shape = [2,3],
+    //    strided_slice.end = [10,20] (larger value than actual dim)
+    //    is treated as strided_slice.end = [2,3]
+    int64_t end_dim = value_from_circle_const(end_const, i);
+    if (not(end_dim >= input_node->dim(i).value() or end_mask.test(i) == true))
+      return false;
+
+    int64_t strides_value = value_from_circle_const(strides_const, i);
+    if (strides_value != 1)
+      return false;
+
+    if (shrink_axis_mask.test(i) && input_node->dim(i).value() != 1)
+      return false;
+  }
+
+  // build shape for Reshape op
+  bool found = false;
+  std::vector<uint32_t> shrunk_shape;
+  for (uint32_t i = 0; i < input_rank; i++)
+  {
+    if (input_node->dim(i) == 1 and shrink_axis_mask.test(i))
+      found = true;
+    else
+      shrunk_shape.emplace_back(input_node->dim(i).value());
+  }
+
+  if (not found)
+    return false;
+
+  auto reshape_node = build_reshape(input_node->graph(), ss_node->name(), luci::get_origin(ss_node),
+                                    input_node, shrunk_shape);
+
+  replace(ss_node).with(reshape_node);
+  return true;
+}
+
+} // namespace
+
+namespace luci
+{
+
+/**
+ * BEFORE
+ *
+ *     [CircleNode]  [CircleConst] for begin, end and strides, respectively
+ *          \             /
+ *          [CircleStridedSlice]
+ *                 |
+ *            [CircleNode]
+ *
+ * AFTER
+ *
+ *     [CircleNode]  [CircleConst] for shape
+ *           \             /
+ *          [CircleReshape]
```

Comment:
As I wrote before, this diagram is to show before-after the transformation and `[CircleStridedSlice]`  is not removed so it should be visible in the diagram.

Reply:
If you don't underatand what I've wrote, please look at other Pass diagrams.